integration/tests/cook/test_client.py

```diff
@@ -44,6 +46,7 @@ def test_query(self):
         self.assertAlmostEqual(job.cpus, 0.5)
         self.assertAlmostEqual(job.mem, 1.0)
         self.assertEqual(job.max_retries, 5)
+        self.client.kill(uuid)
```

Comment:
If the assertion fails, this won't get called. Should we do this in the `finally` block of a `try`/`finally`?